### PR TITLE
circuit breaker supported probe number

### PR DESCRIPTION
**Pull request by sanxun0325**
Describe what this PR does / why we need it
circuit breaker support probes num

Does this pull request fix one issue?
fix:#411

Describe how you did it
Provides the probe count configuration to match the probe count when the circuit breaker is half-open

Describe how to verify it
Special notes for reviews

### CODE REVIEW

File: core/circuitbreaker/circuit_breaker.go [hunk 1]
```diff
@@ -318,9 +335,12 @@ func (b *slowRtCircuitBreaker) OnRequestComplete(rt uint64, _ error) {
 			// fail to probe
 			b.fromHalfOpenToOpen(1.0)
 		} else {
-			// succeed to probe
-			b.fromHalfOpenToClosed()
-			b.resetMetric()
+			b.addCurProbeNum()
+			if b.probeNumber == 0 || atomic.LoadUint64(&b.curProbeNumber) == b.probeNumber {
```
Comment (sczyh30) [inline core/circuitbreaker/circuit_breaker.go:339]: Are there any circumstances that `curProbeNumber` may exceed `probeNumber`?

File: core/circuitbreaker/circuit_breaker.go [hunk 2]
```diff
@@ -498,8 +521,11 @@ func (b *errorRatioCircuitBreaker) OnRequestComplete(_ uint64, err error) {
 	}
 	if curStatus == HalfOpen {
 		if err == nil {
-			b.fromHalfOpenToClosed()
-			b.resetMetric()
+			b.addCurProbeNum()
+			if b.probeNumber == 0 || atomic.LoadUint64(&b.curProbeNumber) == b.probeNumber {
```
Comment (sczyh30) [inline core/circuitbreaker/circuit_breaker.go:525]: Ditto

File: core/circuitbreaker/circuit_breaker.go (hunk 1 above)
Comment (sanxun0325) [reply core/circuitbreaker/circuit_breaker.go]: modified

File: core/circuitbreaker/circuit_breaker.go (hunk 2 above)
Comment (sanxun0325) [reply core/circuitbreaker/circuit_breaker.go]: modified

**Review by sczyh30 — APPROVED** [review]
LGTM

**Comment by sczyh30** [comment]
@louyuting @luckyxiaoqiang @liqiangz Any other comments?

**Comment by sczyh30** [comment]
Nice work. Thanks for contributing!

**Pull request merged**